### PR TITLE
perf: Do a single overpass request in city mode

### PR DESCRIPTION
**Pull request by JulesL2**
## Description

Create a single overpass query separated with "count" statement and do a single download from overpass API

## Corresponding Github Issues / Pull Requests

https://github.com/ThomasParistech/pretty-gpx/issues/15

Time to get a plot with roads and rivers is now around 15 seconds. 

## Remarks

Change not applied on the mountain mode, there is still the old overpass query function

### CODE REVIEW

**Comment by JulesL2** [comment]
@ThomasParistech  PR ready to review

File: .devcontainer/requirements.txt [hunk 1]
```diff
@@ -18,3 +18,5 @@ ruff==0.4.6
 textalloc==1.1.3
```
Comment (ThomasParistech) [inline .devcontainer/requirements.txt:18]: Good job @JulesL2  :fire: 

I've just introduced a profiler in this PR https://github.com/ThomasParistech/pretty-gpx/pull/20 :zap: 
Can you use it to benchmark your PR against main and provide the screenshots ?
Thanks

**Comment by JulesL2** [comment]
@ThomasParistech 

Old method
Capture d'écran 2024-09-27 094355

New method
Capture d'écran 2024-09-27 094026


A picture is worth a thousand words : big improvements 🔝  🥳

File: pretty_gpx/common/data/overpass_request.py [hunk 2]
```diff
@@ -1,31 +1,198 @@
...
 
 
+@dataclass
+class OverpassQuery:
+    """Class to store all queries needed, to launch them and to split the result."""
+    query_dict: dict[str,str]
+    query_unprocessed_results: dict[str,Result]
+    query_cached_results: dict[str,str]
+
+    def __init__(self) -> None:
```
Comment (ThomasParistech) [inline pretty_gpx/common/data/overpass_request.py:35]: skip this init and add default empty fields with ` = field(default_factory=dict)` ?

File: pretty_gpx/common/data/overpass_request.py [hunk 3]
```diff
@@ -1,31 +1,198 @@
...
+
+    @profile
+    def launch_queries(self) -> None:
+        # The code that query the request is inspired from https://github.com/mocnik-science/osm-python-tools
+        # in the OSMPythonTools/internal/cacheObject.py file
+        # Then we split the result into the multiple results and then process it as it is done with 
+        # overpy.Result. The second part of the code of this function used to process the results
+        # comes from overpy.Overpass.from_json function and has been adapted to this situation with multiple results
+        """Merge all queries into a single one, launch the query, get the results and process them."""
+        if len(self.query_dict.keys()) > 0:
```
Comment (ThomasParistech) [inline pretty_gpx/common/data/overpass_request.py:112]:
```
if len(self.query_dict.keys()) == 0:
   return None
```

to avoid too many indents

**Review by ThomasParistech — APPROVED** [review]
I've also tested it on my machine, and it's really fast!! :zap:

**Comment by JulesL2** [comment]
@ThomasParistech The two conversations are resolved, ready to merge

**Pull request merged**